index.d.ts

```diff
@@ -10,42 +10,116 @@ declare module NpmCheck {
-    regError: any;                       // error communicating with the registry 
-    pkgError: any;                       // error reading the package.json 
-    latest: string;                      // latest according to the registry. 
-    installed: string;                   // version installed 
-    isInstalled: boolean;                // Is it installed? 
-    notInstalled: boolean;               // Is it installed? 
-    packageWanted: string;               // Requested version from the package.json. 
-    packageJson: string;                 // Version or range requested in the parent package.json. 
-    devDependency: boolean;              // Is this a devDependency? 
-    usedInScripts: undefined | string[], // Array of `scripts` in package.json that use this module. 
-    mismatch: boolean;                   // Does the version installed not match the range in package.json? 
-    semverValid: string;                 // Is the installed version valid semver? 
-    easyUpgrade: boolean;                // Will running just `npm install` upgrade the module? 
-    bump: INpmVersionBumpType;           // What kind of bump is required to get the latest 
-    unused: boolean;                     // Is this module used in the code? 
+    /**
+     * name of the module.
+     */
+    moduleName: string;
+
+    /**
+     * url to the home page.
+     */
+    homepage: string;
+
+    /**
+     * error communicating with the registry
+     */
+    regError: any;
+
+    /**
+     * error reading the package.json
+     */
+    pkgError: any;
+
+    /**
+     * latest according to the registry.
+     */
+    latest: string;
+
+    /**
+     * version installed
+     */
+    installed: string;
+
+    /**
+     * Is it installed?
+     */
+    isInstalled: boolean;
+    /**
+     * Is it installed?
+     */
+    notInstalled: boolean;
+
+    /**
+     * Requested version from the package.json.
+     */
+    packageWanted: string;
+
+    /**
+     * Version or range requested in the parent package.json.
+     */
+    packageJson: string;
+
+    /**
+     * Is this a devDependency?
+     */
+    devDependency: boolean;
+
+    /**
+     * Array of `scripts` in package.json that use this module.
+     */
+    usedInScripts: undefined | string[];
+
+    /**
+     * Does the version installed not match the range in package.json?
+     */
+    mismatch: boolean;
+
+    /**
+     * Is the installed version valid semver?
+     */
+    semverValid: string;
+
+    /**
+     * Will running just `npm install` upgrade the module?
+     */
+    easyUpgrade: boolean;
+
+    /**
+     * What kind of bump is required to get the latest
+     */
+    bump: INpmVersionBumpType;
+
+    /**
+     * Is this module used in the code?
+     */
+    unused: boolean;
   }
 
   //The default function returns a promise
-  export default function(options: INpmCheckOptions): {
+  export default function (options: INpmCheckOptions): {
```

Comment:
This is no longer equivalent, since the previous export used the CJS syntax.

We need to use the same syntax here to work as before:

```suggestion
  export = function (options: INpmCheckOptions): {
```

Also, I think that we should remove the `declare module` block altogether. It only causes problems, and since this file is shipped next to `index.js` I don't see any reason why to have it?